tools/cpp/osx_cc_configure.bzl

```diff
@@ -83,6 +84,32 @@ def _compile_cc_file(repository_ctx, src_name, out_name):
              "https://github.com/bazelbuild/bazel/issues with the following:\n" +
              error_msg)
 
+def _compile_cc_file(repository_ctx, src_name, out_name):
+    env = repository_ctx.os.environ
+    xcrun_result = repository_ctx.execute([
+        "env",
+        "-i",
+        "DEVELOPER_DIR={}".format(env.get("DEVELOPER_DIR", default = "")),
+        "xcrun",
+        "--sdk",
+        "macosx",
+        "clang",
+        "-mmacosx-version-min=10.9",
+        "-std=c++11",
+        "-lc++",
+        "-arch",
+        "arm64",
+        "-arch",
+        "x86_64",
+        "-O3",
+        "-o",
+        out_name,
+        src_name,
+    ], 30)
```

Comment:
Here's what works for us:

```starlark
    repository_ctx.execute([
        "env",
        "-i",
        "DEVELOPER_DIR={}".format(env.get("DEVELOPER_DIR", default = "")),
        "xcrun",
        "--sdk",
        "macosx",
        "clang",
        "-mmacosx-version-min=10.9",
        "-std=c++11",
        "-lc++",
        "-O3",
        "-arch",
        "arm64",
        "-arch",
        "x86_64",
        "-Wl,-no_adhoc_codesign",
        "-o",
        out_name,
        src_name,
    ], 30)
    repository_ctx.execute([
        "env",
        "-i",
        "codesign",
        "--force",
        "--sign",
        "-",
        out_name,
    ], 30)
```

Reply:
is the second codesign reproducible here?

Reply:
Reproducible with the same machine type (Intel vs Intel, M1 vs M1).

Reply:
updated with what we learned here